### PR TITLE
Eliminate load cue shim from track object

### PR DESCRIPTION
**Pull request by uklotzde**
I was not able to reproduce the issues reported [here](https://github.com/mixxxdj/mixxx/pull/1242#issuecomment-473098622) yet. But this might help by ensuring that the load cue always exists as its own QObject to keep the set of cue points consistent and complete.

- Ensure that the track always has a proper load cue
- Initialize cues before adding to track to reduce number of emitted signals

### CODE REVIEW

**Comment by uklotzde** [comment]
One minor change: Updating the load cue during analysis is optional. Only the ability to modify the intro or outro cue affects the decision if the analysis is actually started. The load cue must not be considered at this point.

File: src/track/track.cpp [hunk 1]
```diff
@@ -720,65 +722,68 @@ void Track::setWaveformSummary(ConstWaveformPointer pWaveform) {
...
+    CuePointer pLoadCue = findCueByType(Cue::LOAD);
+    VERIFY_OR_DEBUG_ASSERT(pLoadCue) {
+        return kDefaultLoadCuePosition;
+    }
+    return pLoadCue->getCuePosition();
 }
 
-CuePointer Track::createAndAddCue() {
+CuePointer Track::createAndAddCue(Cue::CueType type, CuePosition position, double length) {
+    DEBUG_ASSERT(cueType != Cue::CueType::LOAD);
```
Comment (ninomp) [inline src/track/track.cpp]: Shouldn't this be `type` instead of `cueType`?

**Comment by uklotzde** [comment]
Yes, of course. I didn't notice that I was still on a release build all the time!!

**Comment by ninomp** [comment]
@uklotzde Can I merge this now or are you planning to do more work in this PR?

**Comment by uklotzde** [comment]
No, it is ready. I just needed to fix the bugs that I introduced while transforming the code :see_no_evil: 

Now just a single signal should be sent **after** a cue point has been modified instead of multiple signals with inconsistent temporary states while modifying it. Unfortunately this doesn't fix the remaining issues, but might be helpful to trace the cause.

**Comment by ninomp** [comment]
OK Thank you very much!

**Pull request merged**